code/modules/admin/view_variables/debug_variable_appearance.dm

```diff
@@ -0,0 +1,212 @@
+				value = appearance:mouse_drop_zone
+			if(NAMEOF(appearance, mouse_opacity))
+				value = appearance.mouse_opacity
+			if(NAMEOF(appearance, name))
+				value = appearance.name
+			if(NAMEOF(appearance, opacity))
+				value = appearance.opacity
+			if(NAMEOF(appearance, overlays))
+				value = appearance.overlays
+			if("override") // only /image has this
+				var/image/image_appearance = appearance
+				value = image_appearance.override
+			if(NAMEOF(appearance, pixel_x))
+				value = appearance.pixel_x
+			if(NAMEOF(appearance, pixel_y))
+				value = appearance.pixel_y
+			if(NAMEOF(appearance, pixel_w))
+				value = appearance.pixel_w
+			if(NAMEOF(appearance, pixel_z))
+				value = appearance.pixel_z
+			if(NAMEOF(appearance, plane))
+				value = appearance.plane
+			if(NAMEOF(appearance, render_source))
+				value = appearance.render_source
+			if(NAMEOF(appearance, render_target))
+				value = appearance.render_target
+			if(NAMEOF(appearance, suffix))
+				value = appearance.suffix
+			if(NAMEOF(appearance, text))
+				value = appearance.text
+			if(NAMEOF(appearance, transform))
+				value = appearance.transform
+			if(NAMEOF(appearance, underlays))
+				value = appearance.underlays
+
+			if(NAMEOF(appearance, parent_type))
+				value = appearance.parent_type
+			if(NAMEOF(appearance, type))
+				value = "/appearance (as [appearance.type])" // don't fool people
+
+			// These are not documented ones but trackable values. Maybe we'd want these.
+			if(NAMEOF(appearance, animate_movement))
+				value = appearance.animate_movement
+			if(NAMEOF(appearance, dir))
+				value = appearance.dir
+			if(NAMEOF(appearance, glide_size))
+				value = appearance.glide_size
+			if("pixel_step_size")
+				value = "" //atom_appearance.pixel_step_size
+				// DM compiler complains this
+
+			// I am not sure if these will be ever detected, but I made a connection just in case.
+			if(NAMEOF(appearance, contents)) // It's not a thing, but I don't believe how DM will change /appearance in future.
+				value = appearance.contents
+			if(NAMEOF(appearance, loc)) // same reason above
+				value = appearance.loc
+			if(NAMEOF(appearance, vis_contents)) // same reason above
+				value = appearance.vis_contents
+			if(NAMEOF(appearance, vis_flags)) // DM document says /appearance has this, but it throws error
+				value = appearance.vis_flags
+
+			// we wouldn't need these, but let's these trackable anyway...
+			if(NAMEOF(appearance, density))
+				value = appearance.density
+			if(NAMEOF(appearance, screen_loc))
+				value = appearance.screen_loc
+			if(NAMEOF(appearance, verbs))
+				value = appearance.verbs
+			if(NAMEOF(appearance, tag))
+				value = appearance.tag
+
+			else
+				return "<li style='backgroundColor:white'>(READ ONLY) [var_name] <font color='blue'>(Undefined var name in switch)</font></li>"
+	catch
+		return "<li style='backgroundColor:white'>(READ ONLY) <font color='blue'>[var_name] = (untrackable)</font></li>"
```

Comment:
> @EvilDragonfiend If you really think we need the `try/catch` pattern then split out the entire `switch()` chain into a new proc because otherwise the indentation is just too ugly for a mile-long proc (the number of switch/if statements obfuscates the true nature of the proc), and then invoke that in the `try` block. so
> 
> ```dm
> /proc/proc_a(variable_name)
>  try
>   value = locate_variable(variable_name)
>  catch
>   //whatever
> 
> /proc/locate_variable(variable_name)
>  switch(variable_name)
>   if(x)
>    return whatever
> ```
> 
> we do incur a little bit more vertical footprint but it's worth it to lessen the amount of horizontal footprint (which is far uglier). also the `locate_variable()` proc is more re-usable in case that matters in the future.

I don't think it's possible because `locate_variable()` is supposed to return, but it doesn't know the difference between 'null' or it should return this error. only try catch knows the difference.

Reply:
oh wait, nevermind. it handles the whole stack. I will fix that too then.